pkg/apiserver/controller/SnapshotController.go

```diff
@@ -0,0 +1,110 @@
+package controller
+
+import (
+	"github.com/gin-gonic/gin"
+	hwameistorapi "github.com/hwameistor/hwameistor/pkg/apiserver/api"
+	"github.com/hwameistor/hwameistor/pkg/apiserver/manager"
+	log "github.com/sirupsen/logrus"
+	"net/http"
+	"strconv"
+)
+
+type ISnapshotController interface {
+	SnapshotList(ctx *gin.Context)
+	SnapshotGet(ctx *gin.Context)
+}
+
+type SnapshotController struct {
+	m *manager.ServerManager
+}
+
+func NewSnapshotController(m *manager.ServerManager) ISnapshotController {
+	log.Info("NewVolumeController start")
+
+	return &SnapshotController{m}
+}
+
+// SnapshotList godoc
+// @Summary 摘要 全局快照列表展式
+// @Description list Snapshot
+// @Tags        Snapshot
+// @Param       page query int32 true "page"
+// @Param       pageSize query int32 true "pageSize"
+// @Param       snapshotName query string false "snapshotName"
+// @Param       state query string false "state"
+// @Param       volumeName query string false "volumeName"
+// @Accept      json
+// @Produce     json
+// @Success     200 {object}  api.SnapshotList
+// @Router      /cluster/snapshots [get]
+func (v *SnapshotController) SnapshotList(ctx *gin.Context) {
+	var failRsp hwameistorapi.RspFailBody
+
+	// 获取path中的page
+	page := ctx.Query("page")
+	// 获取path中的pageSize
+	pageSize := ctx.Query("pageSize")
+	// 获取path中的snapshotName
+	snapshotName := ctx.Query("snapshotName")
+	log.Infof("SnapshotList snapshot = %v", snapshotName)
+
+	// 获取path中的state
+	state := ctx.Query("state")
+	// 获取path中的volumeName
+	volumeName := ctx.Query("volumeName")
+
+	p, _ := strconv.ParseInt(page, 10, 32)
+	ps, _ := strconv.ParseInt(pageSize, 10, 32)
+
+	var queryPage hwameistorapi.QueryPage
+	queryPage.Page = int32(p)
+	queryPage.PageSize = int32(ps)
+	queryPage.SnapshotName = snapshotName
+	queryPage.SnapshotState = hwameistorapi.VolumeStatefuzzyConvert(state)
+	queryPage.VolumeName = volumeName
+
+	lvs, err := v.m.SnapshotController().ListLocalSnapshot(queryPage)
+	if err != nil {
+		failRsp.ErrCode = 500
```

Comment:
there is defination about http status at package `http`, use that may be better

Reply:
Has been modified